pkg/informer/informer.go

```diff
@@ -87,33 +138,31 @@ type DynamicDiscoverySharedInformerFactory struct {
 	subscribers     map[string]chan<- struct{}
 }
 
-// NewDynamicDiscoverySharedInformerFactory returns a factory for shared
-// informers that discovers new types and informs on updates to resources of
-// those types.
-func NewDynamicDiscoverySharedInformerFactory(
-	cfg *rest.Config,
+// NewGenericDiscoveringDynamicSharedInformerFactory returns is an informer factory that
+// dynamically discovers new types and begins informing on them.
+// It is a generic implementation for both logical-cluster-aware and
+// logical-cluster-unaware (== single cluster, standard kube) informers.
+func NewGenericDiscoveringDynamicSharedInformerFactory[Informer cache.SharedIndexInformer, Lister genericListerBase, GenericInformer genericInformerBase[Informer, Lister]](
```

Comment:
Would just call it DynamicSharedInformerFactory

Reply:
We know `DynamicSharedInformerFactory` is already something in Kube: https://pkg.go.dev/k8s.io/client-go@v0.24.4/dynamic/dynamicinformer#DynamicSharedInformerFactory.
And if I'm not mistaken it is `Dynamc` because is provides informers for dynamic resources (backed by unstructured objects). However it doesn't "discover" the available GVRs provided by the server, and assumes all the GVRs requested by `ForResource()` are provided by the server. If it is not the case the retrieved Informer will never be Synced.

The `DiscoveringDynamicSharedInformerFactory`does something more. It maintains informers for a discovered list of GVRs known to be exposed by the server. That's why we renamed it **Discovering**DynamicSharedInformerFactory.

I agree it's probably not the best name, but giving the same name as something well-defined in Kube seems a bit weird to me as well.